Python/Product/PythonTools/PythonTools/Project/PytestConfigureInfoBar.cs

```diff
@@ -0,0 +1,222 @@
+                    Action = action,
+                    Context = InfoBarContext
+                }
+            );
+        }
+
+        protected abstract void SetProperty(string propertyName, string propertyValue);
+
+        protected TestFrameworkType GetTestFramework(string propertyValue) {
+            if (Enum.TryParse(
+                propertyValue,
+                ignoreCase: true,
+                out TestFrameworkType parsedFramework)
+            ) {
+                return parsedFramework;
+            }
+
+            return TestFrameworkType.None;
+        }
+    }
+
+    internal sealed class ConfigurePytestProjectInfoBar : ConfigurePytestInfoBar {
+        private PythonProjectNode Project { get; }
+
+        public ConfigurePytestProjectInfoBar(IServiceProvider site, PythonProjectNode projectNode) : base(site) {
+            Project = projectNode ?? throw new ArgumentNullException(nameof(projectNode));
+        }
+
+        public override async Task CheckAsync() {
+            await CheckAsync(
+                Project.ActiveInterpreter,
+                GetTestFramework(Project.GetProjectProperty(PythonConstants.TestFrameworkSetting, false)),
+                Project.GetProjectProperty(PythonConstants.SuppressPytestConfigPrompt).IsTrue(),
+                Project.GetPyTestConfigFilePath(),
+                Project.Caption,
+                InfoBarContexts.Project
+            );
+        }
+
+        protected override void SetProperty(string propertyName, string propertyValue) {
+            Project.SetProjectProperty(propertyName, propertyValue);
+        }
+    }
+
+    internal sealed class ConfigurePytestWorkspaceInfoBar : ConfigurePytestInfoBar {
+        private IPythonWorkspaceContext WorkspaceContext { get; }
+
+        public ConfigurePytestWorkspaceInfoBar(IServiceProvider site, IPythonWorkspaceContext pythonWorkspaceContext) : base(site) {
+            WorkspaceContext = pythonWorkspaceContext ?? throw new ArgumentNullException(nameof(pythonWorkspaceContext));
+        }
+
+        public override async Task CheckAsync() {
+            await CheckAsync(
+                WorkspaceContext.CurrentFactory,
+                GetTestFramework(WorkspaceContext.GetStringProperty(PythonConstants.TestFrameworkSetting)),
+                WorkspaceContext.GetBoolProperty(PythonConstants.SuppressPytestConfigPrompt) ?? false,
+                GetPyTestConfigFilePath(),
+                WorkspaceContext.WorkspaceName,
+                InfoBarContexts.Workspace
+            );
+        }
+
+        private string GetPyTestConfigFilePath() {
+            return PythonConstants.PyTestFrameworkConfigFiles
+                .FirstOrDefault(fileName => File.Exists(PathUtils.GetAbsoluteFilePath(WorkspaceContext.Location, fileName)));
+        }
+
+        protected override void SetProperty(string propertyName, string propertyValue) {
+            //TODO Do I have to await here? Can I just fire and forget? 
```

Comment:
since you are the one defining SetProperty in the base class, make it `async Task SetPropertyAsync` so you can await here.